exporter/trace/cloudtrace_test.go

```diff
@@ -16,94 +16,29 @@ package trace_test
 
 import (
 	"context"
-	"flag"
-	"log"
-	"net"
-	"os"
 	"regexp"
 	"sync"
 	"testing"
 	"time"
 
-	emptypb "github.com/golang/protobuf/ptypes/empty"
+	"github.com/googleinterns/cloud-operations-api-mock/cloudmock"
 	"github.com/stretchr/testify/assert"
-	"google.golang.org/api/option"
-	tracepb "google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
-	"google.golang.org/grpc"
 
 	texporter "github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/trace"
 
 	"go.opentelemetry.io/otel/api/global"
 	export "go.opentelemetry.io/otel/sdk/export/trace"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
-)
-
-type mockTraceServer struct {
-	tracepb.TraceServiceServer
-	mu            sync.Mutex
-	spansUploaded []*tracepb.Span
-	delay         time.Duration
-	// onUpload is called every time BatchWriteSpans is called
-	onUpload func(ctx context.Context, spans []*tracepb.Span)
-}
-
-func (s *mockTraceServer) BatchWriteSpans(ctx context.Context, req *tracepb.BatchWriteSpansRequest) (*emptypb.Empty, error) {
-	var err error
-	if s.onUpload != nil {
-		s.onUpload(ctx, req.Spans)
-	}
-	s.mu.Lock()
-	select {
-	case <-ctx.Done():
-		err = ctx.Err()
-	case <-time.After(s.delay):
-		s.spansUploaded = append(s.spansUploaded, req.Spans...)
-	}
-	s.mu.Unlock()
-	return &emptypb.Empty{}, err
-}
-
-func (s *mockTraceServer) len() int {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	return len(s.spansUploaded)
-}
 
-// clientOpt is the option tests should use to connect to the test server.
-// It is initialized by TestMain.
-var clientOpt []option.ClientOption
-
-var (
-	mockTrace mockTraceServer
+	"google.golang.org/api/option"
+	tracepb "google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
 )
 
-func TestMain(m *testing.M) {
-	flag.Parse()
-
-	serv := grpc.NewServer()
-	tracepb.RegisterTraceServiceServer(serv, &mockTrace)
-
-	lis, err := net.Listen("tcp", "localhost:0")
-	if err != nil {
-		log.Fatal(err)
-	}
-	go func() {
-		_ = serv.Serve(lis)
-	}()
-
-	conn, err := grpc.Dial(lis.Addr().String(), grpc.WithInsecure())
-	if err != nil {
-		log.Fatal(err)
-	}
-	clientOpt = []option.ClientOption{option.WithGRPCConn(conn)}
-
-	os.Exit(m.Run())
-}
-
 func TestExporter_ExportSpan(t *testing.T) {
 	// Initial test precondition
-	mockTrace.spansUploaded = nil
-	mockTrace.delay = 0
+	mock := cloudmock.NewCloudMock()
+	defer mock.Shutdown()
+	clientOpt := []option.ClientOption{option.WithGRPCConn(mock.ClientConn())}
```

Comment:
It's nice that you're able to just pass the client connection from mock to client in Go here, but other languages won't have this luxury, as you mentioned in the description

> In the future, we'd like to integrate this into exporters of all languages in order to allow for unified tests.

Will the mock have a fixed port or some other way of discovering the endpoint after starting it?
No need to change anything, just wanted to clarify the mock "contract"

Reply:
Yeah, for other languages we're currently exploring two options. 
1. We have a standalone server mode, similar to [Moto](https://github.com/spulec/moto#stand-alone-server-mode), where the user will spin up the server manually and optionally pass in the address (there's a default address if nothing is passed in). In the exporter test, they'll then point the exporter to the address they choose. After discussions with the Python exporter team, this is what we were planning on doing (they'll spin up the server in their CI and point their exporter to our mock server). 
2. It is possible to call Go functions from other languages (see [here](https://medium.com/learning-the-go-programming-language/calling-go-functions-from-other-languages-4c7d8bcc69bf)), so if we go this route, they could still use this `cloudmock` library to spin up the server, pass the client connection, etc. but this would probably be a bit more difficult than the first option.

Reply:
I would definitely recommend going with option 1 😄

I think we should also consider pros / cons of updating these Go tests to spin up the Fake server out-of-process so its consistent with the other exporters in the future, but this looks good for now.